### DISCUSSION TITLE
Proposed alternative to caseof

### DISCUSSION THREAD

**Discussion by amigalemming** [AccelerateHS/accelerate#544] [General]
Between accelerate-1.2 and accelerate-1.3 `caseof` seems to be gone. However, neither `ChangeLog` nor `git log` gives an idea what do instead.

**Comment by amigalemming** [comment]
I guess, I can replace it by a sequence of `ifThenElse`.

**Reply by tomsmeding** [reply]
`ifThenElse` is polymorphic over `Acc` and `Exp` in order to work with `-XRebindableSyntax`; the "standard" functions are `acond` and `cond`. If you want `caseof` back, you can also just take [the definition from accelerate-1.2](https://hackage.haskell.org/package/accelerate-1.2.0.0/docs/src/Data.Array.Accelerate.Prelude.html#line-2142); it was actually simply a chain of `cond`.

The function was removed in this commit: https://github.com/AccelerateHS/accelerate/commit/359b20d26e2bbbdea0b2877d43c2e12b3ffff69f. (I found this with `git log -p | less`, then search for `caseof` and ignore the first few spurious internal hits.) The intent of the original was to have some kind of equivalent to Haskell's `case`, which is not really a chain of `cond`, but that's the best we had in accelerate-1.2. In 1.3, we have [`match`](https://hackage.haskell.org/package/accelerate-1.3.0.0/docs/Data-Array-Accelerate.html#v:match), which allows proper pattern matching on sum types.
